sapi/cli/php_cli_server.c

```diff
@@ -348,18 +348,33 @@ static void append_http_status_line(smart_str *buffer, int protocol_version, int
 	smart_str_appendl_ex(buffer, "\r\n", 2, persistent);
 } /* }}} */
 
-static void append_essential_headers(smart_str* buffer, php_cli_server_client *client, int persistent) /* {{{ */
+static void append_essential_headers(smart_str* buffer, php_cli_server_client *client, int persistent, sapi_headers_struct *sapi_headers) /* {{{ */
 {
 	char *val;
 	struct timeval tv = {0};
+	bool append_date_header = true;
+
+	if (sapi_headers != NULL) {
+		zend_llist_position pos;
+		sapi_header_struct *h = (sapi_header_struct*)zend_llist_get_first_ex(&sapi_headers->headers, &pos);
+		while (h) {
+			if (h->header_len > strlen("Date:")-1) {
+				if (strncasecmp(h->header, "Date:", strlen("Date:")-1) == 0) {
```

Comment:
Same here: just use `strlen("Date:")` without `-1`.
I guess you copied the `-1` from the CGI SAPI. However, that case is different because that uses `sizeof` instead of `strlen`. The `-1` is necessary with `sizeof` because `sizeof` includes the NUL terminator whereas `strlen` does not.

See this small C program for example to see how subtracting 1 gives the wrong behaviour:
```c
#include <stdio.h>
#include <string.h>

int main() {
    printf("%d\n", strncasecmp("Date:", "Date-Received:", strlen("Date:") - 1));
    printf("%d\n", strncasecmp("Date:", "Date-Received:", strlen("Date:")));
    return 0;
}
```

will output:

```
0
13
```

So you can see the wrong matching of the header name in the case of subtracting 1.

Reply:
Ops, ok!

Thank you for your time.